File: Pascal/Task165.pas
program Task165;
var a, b, c, d : real;
x0, x1, x2 : real;
begin
  writeln('Введіть коефіцієнти нерівності a,b,c');
  readln( a, b, c);
  d := b * b - 4 * c;
  if d < 0 then {друга нерівність не має коренів}
    writeln('[', a:8:2, ';+inf)')    
  else
    {d не менше 0, тобто 0, або більше 0}
    if d = 0 then
      begin
        x0 := -b/2;
        if a < x0 then writeln( 'x Є [', a:8:2, ';', 
               x0:8:2, ') U (', x0:8:2, '; +inf)')
        else 
          if a = x0 then writeln('x Є (', a:8:2, '; +inf)')
          else writeln('x Є [', a:8:2, '; +inf)')
      end
    else  {вітка, що виконується лише коли d > 0}
      begin
        x1 := (-b-sqrt(d))/2;   x2 := (-b+sqrt(d))/2;
        if a < x1 then
          writeln('x Є [', a:8:2, ';', x1:8:2, 
                  ')U(', x2:8:2, ';+inf)')
        else 
          if a <= x2 then 
              writeln('x Є (', x2:8:2, ';+inf)')
          else writeln('x Є [', a:8:2, ';+inf)')
      end    
end.

File: C++/Task165.cpp
#include<iostream>
#include<cmath>
using namespace std;
int main()
{
  double a,b,c,d;
  cout<<"Введіть коефіцієнти нерівності a,b,c\n";
  cin >> a >> b >> c;
  d = b * b - 4 * c;
  if (d < 0)//друга нерівність не має коренів
  {
    cout<< "["<<a<<";+inf)\n";  
  }
  else
  {//d не менше 0, тобто 0, або більше 0
    if (d == 0)
    {
      double x0 = -b/2;
      if (a < x0) 
      {
        cout<<"x Є ["<<a<< ";"<< x0 << 
          ")U("<<x0<<";+inf)\n";
      }
      else 
      {
        if(a == x0) cout<<"x Є ("<<a<<";+inf)\n"; 
        else  cout<<"x Є ["<<a<<";+inf)\n";
      }
    }
    else
    {//вітка, що виконується лише коли d > 0
      double x1 = (-b-sqrt(d))/2;
      double x2 = (-b+sqrt(d))/2;
      if (a < x1)
      {
        cout<<"x Є ["<<a<<";" << x1 <<")U("
                     <<x2<<";+inf)\n";
      }
      else{
        if (a <= x2) cout<<"x Є ("<<x2<<";+inf)\n";
        else cout<<"x Є ["<<a<<";+inf)\n";
      }
    }  
  }
  return 0;
}
